base/math.jl

```diff
@@ -698,7 +698,9 @@ end
 @inline ^(x::Float64, y::Integer) = x ^ Float64(y)
 @inline ^(x::Float32, y::Integer) = x ^ Float32(y)
 @inline ^(x::Float16, y::Integer) = Float16(Float32(x) ^ Float32(y))
-@inline ^{p}(x::Float16, ::Type{Val{p}}) = Float16(Float32(x) ^ Val{p})
+@inline ^{p}(x::Float16, ::Type{Val{p}}) = Float16(x ^ p)
+@inline ^{p}(x::Float32, ::Type{Val{p}}) = Float32(x ^ p)
+@inline ^{p}(x::Float64, ::Type{Val{p}}) = Float64(x ^ p)
 
```

Comment:
This used to work, but it was invalidated by #20889.  It should have been changed to:
```jl
@inline literal_pow{p}(::typeof(^), x::Float16, ::Type{Val{p}}) = Float16(literal_pow(typeof(^),Float32(x),Val{p}))
```

We shouldn't be defining `^` methods for `Val` at all any more.